File: sorting_&_searching/insertion_sort.py
from typing import List


def insertion_sort(arr: List) -> List:
    length = len(arr)
    for i in range(1, length):
        item = arr[i]
        j = i - 1
        while j>=0 and arr[j] > item:
            arr[j+1] = arr[j]
            j -= 1
            arr[j+1] = item
    return arr

arr = [5,4,3,2]
print(insertion_sort(arr))

File: sorting/go_/insertion_sort.go
package main

import "fmt"

func InsertionSort(nums []int) []int {
	length := len(nums)
	for i := 1; i < length; i++ {
		item := nums[i]
		j := i - 1
		for j >= 0 && nums[j] > item {
			nums[j+1] = nums[j]
			j--
			nums[j+1] = item
		}
	}
	return nums
}

func main() {
	arr := []int{5, 3, 2, 1}
	fmt.Println(InsertionSort(arr))
}
